internal/cmd/gocdk/serve.go

```diff
@@ -16,10 +16,213 @@ package main
 
 import (
 	"context"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"time"
 
 	"golang.org/x/xerrors"
 )
 
 func serve(ctx context.Context, pctx *processContext, args []string) error {
-	return xerrors.New("not implemented")
+	f := newFlagSet(pctx, "serve")
+	if err := f.Parse(args); xerrors.Is(err, flag.ErrHelp) {
+		return nil
+	} else if err != nil {
+		return usagef("gocdk serve: %w", err)
+	}
+	if f.NArg() > 1 {
+		return usagef("gocdk serve [BIOME]")
+	}
+	biome := f.Arg(0)
+	if biome == "" {
+		biome = "dev"
+	}
+
+	// Check first that we're in a Go module.
+	moduleRoot, err := findModuleRoot(ctx, pctx.workdir)
+	if err != nil {
+		return xerrors.Errorf("gocdk serve: %w", err)
+	}
+	// Start main serve loop.
+	logger := log.New(pctx.stderr, "gocdk: ", log.Ldate|log.Ltime)
+	if err := serveLoop(ctx, pctx, logger, moduleRoot, biome); err != nil {
+		return xerrors.Errorf("gocdk serve: %w", err)
+	}
+	return nil
+}
+
+func serveLoop(ctx context.Context, pctx *processContext, logger *log.Logger, moduleRoot, biome string) error {
+	// Log biome that is being used.
+	logger.Printf("Preparing to serve %s...", biome)
+
+	// Create a temporary build directory.
+	buildDir, err := ioutil.TempDir("", "gocdk-build")
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if err := os.RemoveAll(buildDir); err != nil {
+			logger.Printf("Cleaning build: %v", err)
+		}
+	}()
+
+	// Apply Terraform configuration in biome.
+	if err := apply(ctx, pctx, []string{biome}); err != nil {
+		return err
+	}
+
+	// Build and run the server.
+	alloc := &serverAlloc{
+		exePath: filepath.Join(buildDir, "server"),
```

Comment:
Binary name should be a flag? Is this something that the user should be able to configure once elsewhere, since we're building the same binary independent of biome?

Reply:
OK. It wasn't obvious to me that this was a different build artifact than the one we'd build for "reals". The user will need control over the name of that binary, right? Anyway, this can be resolved outside of this PR.